packages/mco/components/disaster-recovery/app-failover-relocate/failover-relocate-modal-body.tsx

```diff
@@ -147,46 +127,36 @@ export const FailoverRelocateModalBody: React.FC<FailoverRelocateModalBodyProps>
       loadError: drPlacementControlsLoadError,
     } = memoizedDRPlacementControls;
 
-    const drPlacementRules: string[] = React.useMemo(
-      () =>
-        placementRulesLoaded && !placementRulesLoadError
-          ? getFilteredDRPlacementRuleNames(placementRules)
-          : [],
-      [placementRules, placementRulesLoaded, placementRulesLoadError]
+    const plsRuleLoaded = placementRulesLoaded && !placementRulesLoadError;
+    const subsLoaded = subscriptionsLoaded && !subscriptionsLoadError;
+    const drpcLoaded =
+      drPlacementControlsLoaded && !drPlacementControlsLoadError;
+
+    const placementRuleMap: PlacementRuleMap = React.useMemo(
+      () => (plsRuleLoaded ? filterDRPlacementRuleNames(placementRules) : {}),
```

Comment:
"placementRules" can be empty as well, in that case "filterDRPlacementRuleNames" will return "undefined"
```suggestion
      () => (plsRuleLoaded && filterDRPlacementRuleNames(placementRules)) || {},
```